plugins/browser-plugin-snowplow-ecommerce/src/ga4/api.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  trackProductView,
+  trackProductListView,
+  trackProductListClick,
+  trackCheckoutStep,
+  trackTransaction,
+  trackAddToCart,
+  trackRemoveFromCart,
+} from '../api';
+import { Currency, GA4EcommerceObject, ItemList, Transaction } from './types';
+import { transformG4ItemsToSPProducts } from './utils';
+
+const DEFAULT_PAYMENT_METHOD = 'unknown';
+
+export function trackGA4ViewPromotion(ecommerce: GA4EcommerceObject) {}
+export function trackGA4SelectPromotion(ecommerce: GA4EcommerceObject) {}
+export function trackGA4ViewCart() {}
+
+export function trackGA4AddToCart(ecommerce: GA4EcommerceObject & Currency, finalCartValue: number) {
```

Comment:
Yeap, as advertised in the accelerator page, this is a 'quick' way to migrate your most important GA ecommerce events to snowplow using these wrappers. If you are at a level of custom contexts on ecommerce events, then I think it is a good place to start the native migration.

_I think this would be the point where the user would like to add their custom dimensions in a context, sent together with the standard snowplow ecommerce attributes._

Kinda torn again, it would be really easy make the second argument an `options` object which would include the required attributes, where needed e.g. `finalCartValue`, and the addition of custom contexts and tracker selection.

Reply:
@jethron 
Added a simple options argument which for now reference the currency where required and any other attribute which is required in some cases. I believe that for now it is good enough for the simple translation layer and the extra contexts etc. could be a ticket to the native implementation.